bin/specextract

```diff
@@ -3060,27 +3060,28 @@ def specextract(args):
                     
             if instrument == "ACIS":
                 try:
-                    if True in [weight == "yes", srcbkg == "bkg" and dobkgresp]: # force background to always be weighted
+                    if True in [srcbkg == "src" and weight == "yes", srcbkg == "bkg" and dobkgresp]: # force background to always be weighted
                         ancrfile, weightfile, fef_file, tdetwmap = create_arf_ext(full_outroot, fullfile, asp_arg, ebin, clobber, verbose, phafile, da_arg, msk_arg, ewmap, bintwmap, wmap_clip, wmap_threshold, pars, tmpdir)
 
                         add_tool_history(ancrfile,toolname,pars,toolversion=__revision__)
 
                     else:
-                        try:
-                            ancrfile, weightfile = create_arf_ps(full_outroot, filename, fullfile, asp_arg, ebin, clobber, verbose, phafile, da_arg, msk_arg, chip_id, skyx, skyy, chipx, chipy)
-
-                            add_tool_history(ancrfile,toolname,pars,toolversion=__revision__)
-
-                        except OSError:
-                            raise IOError("Failed to create ARF for {}".format(fullfile))
+                        if True in [srcbkg == "src", srcbkg == "bkg" and dobkgresp]:
+                            try:
+                                ancrfile, weightfile = create_arf_ps(full_outroot, filename, fullfile, asp_arg, ebin, clobber, verbose, phafile, da_arg, msk_arg, chip_id, skyx, skyy, chipx, chipy)
 
-                        if correct == "yes":
-                            v1("Calculating aperture correction for {0} ARF {1}".format(srcbkg,iteminfostr))
+                                add_tool_history(ancrfile,toolname,pars,toolversion=__revision__)
 
-                            try:
-                                ancrfile = correct_arf(full_outroot, fullfile, filename, ancrfile, skyx, skyy, binarfcorr, clobber)
                             except OSError:
-                                raise IOError("Failed to PSF correct the ARF: {}".format(ancrfile))
+                                raise IOError("Failed to create ARF for {}".format(fullfile))
+
+                            if correct == "yes":
+                                v1("Calculating aperture correction for {0} ARF {1}".format(srcbkg,iteminfostr))
+                                
+                                try:
+                                    ancrfile = correct_arf(full_outroot, fullfile, filename, ancrfile, skyx, skyy, binarfcorr, clobber)
+                                except OSError:
+                                    raise IOError("Failed to PSF correct the ARF: {}".format(ancrfile))
```

Comment:
[Background reading, not a suggested code change]

For situations where we catch an error and then raise an error, I've started to occasionally use the

    raise BlahError(...) from None

syntax, since this cuts down on the huge chain of errors reported to the user. However, in this case we don't want that "optimisation" since it's useful to be able to find the actual error when handling helpdesk tickets and the like.